src/GraphQL/loaders/defaultGraphQLTypes.js

```diff
@@ -1020,6 +1021,56 @@ const SIGN_UP_RESULT = new GraphQLObjectType({
   },
 });
 
+const ElEMENT = new GraphQLObjectType({
+  name: 'Element',
+  description:
+    'The SignUpResult object type is used in the users sign up mutation to return the data of the recent created user.',
+  fields: {
+    value: {
+      description: 'Return the value of the element in the array',
+      type: new GraphQLNonNull(ANY),
+    },
+  },
+});
+
+// Default static union type, we update types and resolveType function later
+const ARRAY_RESULT = new GraphQLUnionType({
```

Comment:
the 'outputFields` are lazy loaded. So you don't need to define it here and change it later. You can just do `let ARRAY_RESULT;` or even instead of declaring here, attach it in the `parseGraphQLSchema` when loaded.

Reply:
`ARRAY_RESULT` is needed in `ParseClassTypes`. 
```js
 case 'Array':
      return new GraphQLList(defaultGraphQLTypes.ARRAY_RESULT);
```
To ensure `ARRAY_RESULT` validity i prefer to set it to a default state and then overwrite the `UnionType` after all `ParseClasses` are loaded

Reply:
Or maybe i missed something in your suggestion

Reply:
This line you sent is inside the `mapOutputType` function (https://github.com/parse-community/parse-server/blob/master/src/GraphQL/loaders/parseClassTypes.js#L59) that is only used by the `outputFields` function (https://github.com/parse-community/parse-server/blob/master/src/GraphQL/loaders/parseClassTypes.js#L537). If you look at https://github.com/parse-community/parse-server/blob/master/src/GraphQL/loaders/parseClassTypes.js#L646 and https://github.com/parse-community/parse-server/blob/master/src/GraphQL/loaders/parseClassTypes.js#L684, both types are returning the fields as a function for lazy loading. It means that all these functions will only be called after all schema is already generated and returned to Apollo Server. So it gives you the ability to define the type only a single time (even knowing that the `parseClassTypes` loader run before.

I know the way you did should also work but changing the type after it was created is something that was not designed by the GraphQL reference implementation and we can have bugs in the future because of this (even higher chances because we are using a "non public" `_types` property).